src/index.ts

```diff
@@ -64,104 +156,192 @@ const dmMachine = createMachine(
+              }, 
+              
+              
+
+
+              CheckSlots: {
+                always: [
+                  { target: 'Askmusic', guard: 'ismusicMissing' },
+                  { target: 'Asksinger', guard: 'issingerMissing' },
+                  { target: 'Asksong', guard: 'issongMissing' },
+                  { target: 'Asktype', guard: 'istypeMissing' },
+                  { target: 'Askvolume', guard: 'isvolumeMissing' },
+                  { target: 'SayBack' }, 
+                ]
                 },
-                on: { SPEAK_COMPLETE: "Ask" },
+              Askmusic: {
+                entry: say('Which music would you like?'),
+                on:{ SPEAK_COMPLETE: 'Ask' }
               },
-              IdleEnd: {},
-            },
+              Asksinger: {
+                entry: say('Which singer would you like?'),
+                on: { SPEAK_COMPLETE: 'Ask' }
+              },
+              Asksong: {
+                entry: say('Which song would you like?'),
+                on: { SPEAK_COMPLETE: 'Ask' }
+              },
+              Asktype: {
+                entry: say('Which type would you like?'),
+                on: { SPEAK_COMPLETE: 'Ask' }
+              },
+              Askvolume: {
+                entry: say('Which volume would you like?'),
+                on: { SPEAK_COMPLETE: 'Ask' }
+              },
+              SayBack: {
+                entry: 'navigateFeedback',
+                on: {
+                  SPEAK_COMPLETE: {actions: "prepare"}
+                }
+              },
+            }
           },
         },
       },
-      GUI: {
-        initial: "PageLoaded",
-        states: {
-          PageLoaded: {
-            entry: "gui.PageLoaded",
-            on: { CLICK: { target: "Inactive", actions: "prepare" } },
+
+      AskChatGPT: {
+        invoke: {
+            src: fromPromise(async ({ input }) => {
+                const data = await fetchFromChatGPT(
+                    input.lastResult[0].utterance + "give it to me in a json format with entities: music, singer, song, type and volume",
+                    40,
+                );
+                return data;
+            }),
+
+            input: ({ context, event }) => ({
+              lastResult: context.lastResult,
+          }),
+          onDone: {
+              target: "SayBack",
+              actions: [
+                ({ event }) => console.log(JSON.parse(event.output)),
+                assign({ recipeName: ({ event }) => JSON.parse(event.output).recipeName,
+              music: ({event}) => JSON.parse(event.output).prepTime,
+              singer: ({ event }) => JSON.parse(event.output).cookTime,
+              song: ({ event }) => JSON.parse(event.output).servings,
+              type: ({ event }) => JSON.parse(event.output).ingredients,
+              volume: ({ event }) => JSON.parse(event.output).instructions,
+            }),
+          ],
+          },
+          SayBack: {
```

Comment:
This is misplaced: now it is inside `invoke` but should be a sibling of `AskChatGPT`.